pylib/cqlshlib/driver.py

```diff
@@ -0,0 +1,145 @@
+        if host.endpoint in self._allowed_hosts_resolved:
+            RoundRobinPolicy.on_add(self, host)
+
+
+def cluster_factory(host, whitelist_lbp=True, **kwargs):
+    """
+    Cluster factory to create a cassandra or dse cluster for cqlsh.
+
+    :param host: the host for the connection. This can be the hostname string or an EndPoint instance.
+    :param whitelist_lbp: Specify if a WhiteListRoundRobinPolicy should be applied. This can be set to True,
+                          False or an EndPoint instance, which is used to configure the policy when
+                          connecting to a cloud cluster. On a cloud cluster, the LBP has to be set after the
+                          nodes discovery. Otherwise, the LBP is set as usual to the execution profile.
+                          Default to True.
+    :kwargs: All other keyword arguments are passed to the Cluster constructor.
+    :return: A Cluster instance.
+    """
+
+    is_cloud_cluster = False
+    is_unix_socket_endpoint = False
+    endpoint = host
+    options = kwargs.copy()
+
+    # Configure the cluster contact point type and address.
+    if is_unix_socket(host):
+        # update endpoint and load balancing policy for unix socket
+        endpoint = UnixSocketEndPoint(host)
+        is_unix_socket_endpoint = True
+
+    # Determine if we are trying to connect to a cloud cluster
+    secure_connect_bundle = options.pop('secure_connect_bundle', None)
+    if secure_connect_bundle:
+        is_cloud_cluster = True
+        options['cloud'] = {'secure_connect_bundle': secure_connect_bundle}
+    elif 'cloud' in options:
+        is_cloud_cluster = options['cloud'] is not None
+
+    # Build the Cluster instance
+    if is_cloud_cluster:
+        return _cloud_cluster_factory(whitelist_lbp, **options)
+
+    if is_unix_socket_endpoint and 'port' in options:
+        del options['port']
+
+    contact_points = (endpoint,)
+
+    if whitelist_lbp:
+        lbp_class = WhiteListRoundRobinPolicy
+        if is_unix_socket_endpoint:
+            lbp_class = EndpointWhiteListRoundRobinPolicy
+
+        whitelist = [endpoint] if whitelist_lbp is True else [whitelist_lbp]
+        options['load_balancing_policy'] = lbp_class(whitelist)
+
+    return Cluster(contact_points=contact_points, **options)
+
+
+def is_unix_socket(hostname):
+    if isinstance(hostname, six.string_types) and os.path.exists(hostname):
+        mode = os.stat(hostname).st_mode
+        return stat.S_ISSOCK(mode)
+    return False
+
+
+def _cloud_cluster_factory(whitelist_lbp, **kwargs):
+    """
+    Create cloud cluster from given options.
+    Please notice that:
+    - cloud should be present in option
+    - contact_points, endpoint_factory, ssl_context, and ssl_options cannot be specified with a cloud configuration (will be removed)
+    - whitelist_lbp can be True (random contact point), False (no policy set), or specific endpoint
+    """
+    options = kwargs.copy()
+    assert 'cloud' in options
```

Comment:
L124-128 are for fixing DSE existing `LOGIN` issue when too many parameters are set together with `cloud`.
In this function we are creating *cloud* connection and cloud connection only.
I am quite surprised that there are no dedicated python driver factory methods to:
- create cloud connection with only accepted parameters
- login: create clone of connection just with different auth